src/_pytest/assertion/util.py

```diff
@@ -308,6 +311,21 @@ def _diff_text(left: str, right: str, verbose: int = 0) -> list[str]:
                 ]
                 left = left[:-i]
                 right = right[:-i]
+        shortest = min(left, right, key=lambda x: len(x))
+        lines = j = start = 0
+        if shortest.count("\n") >= DEFAULT_MAX_LINES:
```

Comment:
I'm afraid this PR got outdated after https://github.com/pytest-dev/pytest/pull/12766.

I think the path forward would be for `_diff_text()` to receive both "max lines" and "max chars" by parameter instead (they being `int | None`, with `None` meaning "no limits").